apps/challenges/aws_utils.py

```diff
@@ -689,17 +689,53 @@ def restart_workers_signal_callback(sender, instance, field_name, **kwargs):
     prev = getattr(instance, "_original_{}".format(field_name))
     curr = getattr(instance, "{}".format(field_name))
     if prev != curr:
+        challenge = None
         if field_name == "test_annotation":
             challenge = instance.challenge
         else:
             challenge = instance
-        restart_workers([challenge])
+
+        response = restart_workers([challenge])
+        count, failures = response["count"], response["failures"]
+
         logger.info(
             "The worker service for challenge {} was restarted, as {} was changed.".format(
-                instance.pk, field_name
+                challenge.pk, field_name
             )
         )
 
+        if (count != 1):
```

Comment:
what happens in case the challenge has more than 1 workers? can we fetch the total number of workers to compare from models?

Reply:
Here, count refers to the number of challenges actually. It's due to the start_workers method implementation. So, it should be 1 since we're doing this only for 1 challenge.